sql/analyzer/indexed_joins.go

```diff
@@ -651,13 +655,93 @@ func addHashJoins(m *memo.Memo) error {
 	})
 }
 
+func addSlidingRangeJoin(m *memo.Memo) error {
+	return memo.DfsRel(m.Root(), func(e memo.RelExpr) error {
+		switch e.(type) {
+		case *memo.InnerJoin, *memo.LeftJoin:
+		default:
+			return nil
+		}
+
+		join := e.(memo.JoinRel).JoinPrivate()
+
+		_, lIndexes, lFilters := lookupCandidates(join.Left.First)
+		_, rIndexes, rFilters := lookupCandidates(join.Right.First)
+
+		for _, filter := range join.Filter {
+
+			switch f := filter.(type) {
+			case *memo.Between:
+				if !(satisfiesScalarRefs(f.Value.Scalar, join.Left) &&
+					satisfiesScalarRefs(f.Min.Scalar, join.Right) &&
+					satisfiesScalarRefs(f.Max.Scalar, join.Right)) {
+					return nil
+				}
+				// TODO: Is this safe? If the expression references multiple columns, does this reference one
+				// arbitrarily?
+				valueColRef := getColumnRefFromScalar(f.Value.Scalar)
+				minColRef := getColumnRefFromScalar(f.Min.Scalar)
+				maxColRef := getColumnRefFromScalar(f.Max.Scalar)
+				if valueColRef == nil || minColRef == nil || maxColRef == nil {
+					return nil
+				}
+
+				leftIndexScans := sortedIndexScansForTableCol(lIndexes, valueColRef, join.Left.RelProps.FuncDeps().Constants(), lFilters)
+				if leftIndexScans == nil {
+					leftIndexScans = []*memo.IndexScan{nil}
+				}
+				for _, lIdx := range leftIndexScans {
+					rightIndexScans := sortedIndexScansForTableCol(rIndexes, minColRef, join.Right.RelProps.FuncDeps().Constants(), rFilters)
+					if rightIndexScans == nil {
+						rightIndexScans = []*memo.IndexScan{nil}
+					}
+					for _, rIdx := range rightIndexScans {
+						rel := &memo.SlidingRangeJoin{
+							JoinBase: join.Copy(),
+						}
+						// TODO: Remove the filter that was used to create the sliding range because it's no longer
```

Comment:
you can include a filter in the plan representation that you drop converting to the exec format

Reply:
Removed for now. If a range join uses an index with multiple key expressions, removing the right set of filters is no longer simple.

We can attempt this if it ever proves to be a performance bottleneck, but I doubt it will.